tests/src/PatternAssertions/HighlightedListAssert.php

```diff
@@ -0,0 +1,140 @@
+    self::assertCount(1, $highlighted_list);
+  }
+
+  /**
+   * Asserts the Highlighted item of the list.
+   *
+   * @param array|null $expected_highlighted_item
+   *   The expected item array (image array, title, url, primary meta array
+   *   and secondary meta).
+   * @param string $selector
+   *   The CSS selector to find the item.
+   * @param \Symfony\Component\DomCrawler\Crawler $crawler
+   *   The DomCrawler where to check the element.
+   */
+  protected function assertHighlightedItem(?array $expected_highlighted_item, string $selector, Crawler $crawler) {
+    if (is_null($expected_highlighted_item)) {
+      $this->assertElementNotExists($selector, $crawler);
+      return;
+    }
+    $this->assertElementExists($selector, $crawler);
+    $highlighted_wrapper = $crawler->filter($selector);
+    $list_item_assert = new ListItemAssert();
+    $html = $highlighted_wrapper->html();
+    $list_item_assert->assertPattern($expected_highlighted_item, $html);
+    $list_item_assert->assertVariant('default', $html);
+  }
+
+  /**
+   * Asserts the items on the right side of the highlighted list.
+   *
+   * @param array|null $expected_items
+   *   The expected items.
+   * @param string $selector
+   *   The CSS selector to find the items' wrapper.
+   * @param \Symfony\Component\DomCrawler\Crawler $crawler
+   *   The DomCrawler where to check the element.
+   *
+   * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+   */
+  protected function assertItems(?array $expected_items, string $selector, Crawler $crawler): void {
+    if (is_null($expected_items)) {
+      $this->assertElementNotExists($selector, $crawler);
+      return;
+    }
+    $this->assertElementExists($selector, $crawler);
+    $items_wrapper = $crawler->filter($selector);
+    // Assert the number of items.
+    $items = $items_wrapper->filter('article');
+    self::assertCount(count($expected_items), $items, 'The expected number of items does not correspond with the actual number of items in the list.');
+    // Assert there is no description element for any of the items.
+    $this->assertElementNotExists('.ecl-content-block__description', $items_wrapper);
+
+    // Assert each item's info.
+    $list_item_assert = new ListItemAssert();
+    foreach ($expected_items as $index => $expected_item) {
+      $html = $expected_item->html();
+      $list_item_assert->assertPattern($expected_item, $html);
+      $list_item_assert->assertVariant('default', $html);
+    }
+  }
+
+  /**
+   * Asserts the detail text from the footer of the highlighted list.
+   *
+   * @param array|null $expected
+   *   The expected description values.
+   * @param string $selector
+   *   The CSS selector to find the detail.
+   * @param \Symfony\Component\DomCrawler\Crawler $crawler
+   *   The DomCrawler where to check the element.
+   */
+  protected function assertDescription($expected, string $selector, Crawler $crawler): void {
```

Comment:
Is this needed?

Reply:
It should be used for the formatted text in the footer, below the See more link.